src/app/course/course-question-snippet/course-question-snippet.component.html

```diff
@@ -44,6 +44,17 @@ <h2 *ngIf="eventId" class="mb-3 mt-5">{{ event.name }} Questions</h2>
                                 <a class="btn btn-warning mr-3"
                                    [routerLink]="['/problem', uqj.question.id,'edit']">Edit</a>
                             </ng-container>
+                            <ng-container>
+                                <button type="button" mat-icon-button
+                                        [ngClass]="{'red-icon':favoriteStatus[uqj?.question.id], 'gray-icon':!favoriteStatus[uqj?.question.id]}"
```

Comment:
This can just use a conditional variable in the component instead. Since both of these are exclusive